features/step_definitions/sample_file_steps.rb

```diff
@@ -26,7 +26,7 @@ def a
 end
 
 Given(/^the "(.*?)" sample file exists$/) do |file_name|
-  full_path = File.expand_path file_name, 'spec/samples'
+  full_path = Pathname(__dir__) + '../../spec/samples' + file_name
```

Comment:
I wasn’t sure whether there isn’t an edge-case where `Pathname` would escape something properly that concatenation wouldn’t, but I couldn’t come up with anything so I switched them all to interpolated Strings. :)